EcommerceApp/backend/spring-boot-ecommerce-app/src/main/java/kkramarenko/ecommerceapp/service/LoyaltyProgramServiceImpl.java

```diff
@@ -0,0 +1,56 @@
+package kkramarenko.ecommerceapp.service;
+
+import kkramarenko.ecommerceapp.entity.Customer;
+import kkramarenko.ecommerceapp.entity.Order;
+import kkramarenko.ecommerceapp.enums.DiscountRate;
+import kkramarenko.ecommerceapp.repository.OrderRepository;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Service
+public class LoyaltyProgramServiceImpl implements LoyaltyProgramService {
+
+    private final OrderRepository orderRepository;
+
+    public LoyaltyProgramServiceImpl(OrderRepository orderRepository) {
+        this.orderRepository = orderRepository;
+    }
+
+    /**
+     * Used after new purchase for some customer is saved to database
+     *
+     * Checks if sum of all customer orders' totalPrice is more than some value,
+     * return customer's current discount rate based on that
+     * @see DiscountRate
+     *
+     * @param customer - customer to check discount rate
+     * @return DiscountRate
+     */
+    @Override
+    public DiscountRate getCustomerCurrentDiscountRate(Customer customer) {
+
+        List<Order> customerOrders = orderRepository.findOrdersByCustomer(customer);
+
+        if(customerOrders.size() == 0) { return DiscountRate.starter; }
+
+        BigDecimal customerOrdersTotalPriceSum = new BigDecimal("0.00");
+        for(Order order: customerOrders){
+            customerOrdersTotalPriceSum = customerOrdersTotalPriceSum.add(order.getTotalPrice());
+        }
+
+        if(customerOrdersTotalPriceSum.compareTo(new BigDecimal("100.00")) < 0){
```

Comment:
The best place for such constants is database, not the code. Actually it takes effort to implement so there are other options:
for example, you can declare these numbers as named constants or as fields in DiscountRate enum

Reply:
Done